wp-content/themes/twentytwenty-child/functions.php

```diff
@@ -293,4 +338,232 @@ function update_students_settings() {
+add_action( 'wp_ajax_ajax_func', 'ajax_func' );
+
+/** 
+ * Add a column for the activity status on the student dashboard 
+ * */
+function add_student_columns( $columns ) {
+    return array_merge( $columns, array( 'Active' => __( 'Active' ) ) );
+}
+add_filter('manage_student_posts_columns' , 'add_student_columns');
+
+/**
+ * Print the activity checkboxes on the dashboard
+ *  */ 
+function print_extra_columns() { 
+    $meta = get_post_meta( get_the_ID() );
+    $checked = '';
+    if( $meta["status"][0] == 1 ) $checked = 'checked';
+    echo '<form method="post"> <div > <input type="checkbox" class="student-status" id="' . get_the_ID() . '" name="status" value="1" ' . $checked . '  /> </div> </form>';
+}
+add_action( 'manage_student_posts_custom_column', 'print_extra_columns' );
+
+/**
+ * Make the custom column sortable
+ */
+function sort_student_status( $columns ) {
+    $columns['Active'] = __('status');
+    return $columns;
+}
+add_filter( 'manage_edit-student_sortable_columns', 'sort_student_status' );
+
+/**
+ * Define how to order our sortable custom column
+ */
+function student_status_orderby( $query ) {
+    if( ! is_admin() ) return;
+ 
+    $orderby = $query->get( 'orderby' );
+ 
+    if( 'status' == $orderby ) {
+        $query->set('meta_key','status');
+        $query->set('orderby','meta_value');
+    }
+}
+add_action( 'pre_get_posts', 'student_status_orderby' );
+
+/**
+ * Function executed by AJAX to update activity status
+ */
+function update_student_status(){
+    $status = ( esc_attr( $_POST['checked'] ) === 'true' ) ?  1 : 0;
+    update_post_meta( sanitize_text_field( $_POST['student-id'] ), 'status', $status );
+}
+add_action( 'wp_ajax_nopriv_update_student_status', 'update_student_status' );
+add_action( 'wp_ajax_update_student_status', 'update_student_status' );
+
+/** 
+ * Add custom top-level menu page 
+ * */
+function register_dictionary_menu() {
+    add_menu_page(
+        'Dictionary',
+        'Dictionary',
+        'manage_options',
+        'dictionary',
+        'dictionary_callback' );
+}
+add_action( 'admin_menu', 'register_dictionary_menu' );
+
+/**
+ * Callback func for the menu page
+ * */
+function dictionary_callback() {
+    echo '<div> <h2>Dictionary</h2> </div>';
+    settings_fields( 'dictionary' );
+    do_settings_sections( 'dictionary' );
+}
+
+/** 
+ * Add custom settings to the new settings page 
+ * */
+function register_dictionary_settings() {
+    add_settings_section( 'dictionary', '', null, 'dictionary' );
+
+    add_settings_field( 'dictionary_search'   , 'Search up a word: ', 'show_dictionary_search'  , 'dictionary', 'dictionary' );
+
+    register_setting( 'dictionary', 'dictionary_search' );
+}
+add_action( 'admin_init', 'register_dictionary_settings' );
+
+/** 
+ * Callback for a custom setting 
+ * */
+function show_dictionary_search() {
+    $body = !empty(get_transient( 'dictionary_transient' )) ? get_transient( 'dictionary_transient' ) : '';
```

Comment:
You are running get_transient twice here, can we optimize that? Also, formatting.